libraries/exoplayer/src/main/java/androidx/media3/exoplayer/drm/HttpMediaDrmCallback.java

```diff
@@ -182,7 +192,17 @@ private static byte[] executePost(
       while (true) {
         DataSourceInputStream inputStream = new DataSourceInputStream(dataSource, dataSpec);
         try {
-          return Util.toByteArray(inputStream);
+          byte[] response = Util.toByteArray(inputStream);
+          lastLoadEventInfo =
+              new LoadEventInfo(
+                  -1,   // note this is replaced with the actual taskId from the request
```

Comment:
What do you think about plumbing an (optional) `taskId` into `ExoMediaDrm.KeyRequest` and `ExoMediaDrm.ProvisionRequest`? Then it would be available here and you wouldn't need to use this placeholder value.

Reply:
Yes, it is a bit ugly this way...  I stopped short because changing the `MediaDrmCallback.executeProvisionRequest()` signature, as it has lots of implementations (unless we add new methods, which is also kind of clumsy feeling.   What if we added the value to `ProvisionRequest` and `KeyRequest`

Reply:
@icbaker  this may fall into the category of perfect is the enemy of done category, but yes... It looks distatesful.

Reply:
> What if we added the value to `ProvisionRequest` and `KeyRequest`

This is what I originally intended, but I just had a play around with this and it would require adding `taskId` to `ExoMediaDrm.getKeyRequest`, so it also has interface evolution problems (and this would also be the first piece of `KeyRequest` that isn't derived directly (or slightly indirectly) from the framework `MediaDrm.getKeyRequest`.

----

I agree that your current approach is probably the least bad option here.

However unfortunately while thinking about this I spotted a different/bigger problem which I'll raise in a new comment.